include/zephyr/internal/syscall_handler.h

```diff
@@ -394,6 +394,20 @@ int k_usermode_string_copy(char *dst, const char *src, size_t maxlen);
  */
 #define K_SYSCALL_VERIFY(expr) K_SYSCALL_VERIFY_MSG(expr, #expr)
 
+/**
+ * @intenal @brief Macro to check if size is negative
```

Comment:
this is wrong, do not add @internal here.

Reply:
when this should be used then ? Is there any particular case where `@internal` is valid ?